examples/kubernetes/connectivity-check/Makefile

```diff
@@ -1,35 +1,90 @@
 # Copyright 2017-2020 Authors of Cilium
 # SPDX-License-Identifier: Apache-2.0
-#
+
+include ../../../Makefile.defs
+include ../../../Makefile.quiet
+
 DEFAULT_OUT := connectivity-check.yaml
 HOSTPORT_OUT := connectivity-check-hostport.yaml
 SINGLE_OUT := connectivity-check-single-node.yaml
+PROXY_OUT := connectivity-check-proxy.yaml
+QUARANTINE_OUT := connectivity-check-quarantine.yaml
+
+SERVERS_NAME := echo-a echo-b echo-b-host echo-b-host-headless echo-c echo-c-host echo-c-host-headless
+SERVERS_OUT := servers.yaml
 
-SRC := $(wildcard *.yaml)
-HOSTPORT_SRC := $(filter-out $(DEFAULT_OUT) $(HOSTPORT_OUT) $(SINGLE_OUT), $(SRC))
-DEFAULT_SRC := $(filter-out $(wildcard *-hostport.yaml),$(HOSTPORT_SRC))
-SINGLE_SRC := $(filter-out $(wildcard *-multi-node*.yaml),$(DEFAULT_SRC))
+SRC := $(wildcard *.cue)
+ALL_TARGETS := $(DEFAULT_OUT) $(HOSTPORT_OUT) $(SINGLE_OUT) $(PROXY_OUT) $(QUARANTINE_OUT)
 
+DOCKER_RUN := $(CONTAINER_ENGINE) container run --rm \
+	--workdir /src/examples/kubernetes/connectivity-check \
+	--volume $(CURDIR)/../../..:/src \
+	--user "$(shell id -u):$(shell id -g)"
 
-all: $(DEFAULT_OUT) $(HOSTPORT_OUT) $(SINGLE_OUT)
+CUE_IMAGE := "docker.io/cuelang/cue:v0.2.2@sha256:2ea932c771212db140c9996c6fff1d236f3f84ae82add914374cee553b6fc60c"
+CUE := $(DOCKER_RUN) $(CUE_IMAGE)
 
-$(DEFAULT_OUT): $(DEFAULT_SRC)
+all: $(ALL_TARGETS)
+
+$(DEFAULT_OUT): $(SRC)
 	@echo '# Automatically generated by Makefile. DO NOT EDIT' > $(DEFAULT_OUT)
-	for FILE in $(DEFAULT_SRC); do \
-		cat $$FILE >> $(DEFAULT_OUT); \
-		echo "---" >> $(DEFAULT_OUT); \
-	done
+	$(CUE) dump > $(DEFAULT_OUT)
 
-$(HOSTPORT_OUT): $(HOSTPORT_SRC)
+$(HOSTPORT_OUT): $(SRC)
 	@echo '# Automatically generated by Makefile. DO NOT EDIT' > $(HOSTPORT_OUT)
-	for FILE in $(HOSTPORT_SRC); do \
-		cat $$FILE >> $(HOSTPORT_OUT); \
-		echo "---" >> $(HOSTPORT_OUT); \
-	done
+	$(CUE) -t component=all dump > $(HOSTPORT_OUT)
 
-$(SINGLE_OUT): $(SINGLE_SRC)
+$(SINGLE_OUT): $(SRC)
 	@echo '# Automatically generated by Makefile. DO NOT EDIT' > $(SINGLE_OUT)
-	for FILE in $(SINGLE_SRC); do \
-		cat $$FILE >> $(SINGLE_OUT); \
-		echo "---" >> $(SINGLE_OUT); \
+	$(CUE) -t topology=single-node dump > $(SINGLE_OUT)
+
+$(PROXY_OUT): $(SRC) $(SERVERS_OUT)
+	@echo '# Automatically generated by Makefile. DO NOT EDIT' > $(PROXY_OUT)
+	$(CUE) -t component=proxy dump > $@
+	cat $(SERVERS_OUT) >> $@
+
+$(QUARANTINE_OUT): $(SRC) $(SERVERS_OUT)
+	@echo '# Automatically generated by Makefile. DO NOT EDIT' > $(QUARANTINE_OUT)
+	$(CUE) -t component=all -t quarantine=true dump > $@
+	cat $(SERVERS_OUT) >> $@
+
+clean:
+	@rm -f $(ALL_TARGETS)
+	@rm -f *.new
+	@rm -f *.diff
+	@rm -f *.yaml
+
+deploy:
+	kubectl apply -f connectivity-check-hostport.yaml
```

Comment:
should it be common `connectivity-check.yaml` ?

Reply:
Good question. Here I opted for the larger YAML to try to ensure that any developers who use this target will get more information (even checks that may potentially fail depending on cluster configuration), just to reduce the likelihood that we overlook the extra checks in the hostport YAML.